extension/src/data/index.ts

```diff
@@ -68,9 +70,12 @@ export abstract class BaseData<
   private watchFiles() {
     return this.dispose.track(
       createFileSystemWatcher(getRelativePattern(this.dvcRoot, '**'), path => {
+        const relPath = relative(this.dvcRoot, path)
         if (
-          this.getWatchedFiles().some(watchedRelPath =>
-            path.endsWith(watchedRelPath)
+          this.getWatchedFiles().some(
+            watchedRelPath =>
+              path.endsWith(watchedRelPath) ||
+              isSameOrChild(relPath, watchedRelPath)
```

Comment:
[F] If the parent directory has changed then we can assume that the watched path has changed.

Here is the example from `magnetic-tiles-defect`:

Plots folder structure looks like this:

```
└── training
    ├── metrics.json
    └── plots
        └── metrics
            ├── dice.tsv
            ├── eval
            │   └── loss.tsv
            ├── jaccard_coeff.tsv
            └── train
                └── loss.tsv
```

We get only the following event during an experiment:

```
[trace] [File Watcher (parcel)] [ADDED] /workspaces/magnetic-tiles-defect/training/metrics.json
[trace] [File Watcher (parcel)] [ADDED] /workspaces/magnetic-tiles-defect/training/plots
```

That meant that we missed the event because none of the "watched paths" had been changed.